### DISCUSSION TITLE
.raw import : imported but not indexed, lost original file names.

### DISCUSSION THREAD

**Discussion by artincube** [photoprism/photoprism#3554] [Q&A]
Behavior:

On import of an archive of old 5DIII files.
Some are files with .raw extension, I found out after getting the errors from Photoprism.
These are old Magic Lantern raw video files. They basically contain a series of dng frames but probably encoded in a streaming (recording from sensor) format with headers if any that are unknown to Photoprism and its modules.

These files get imported, renamed (!?) in the Photoprism date format, but not indexed (ok) and get errors (ok) as files not recognized, no meaningful info. No yml created in sidecar (!? if file name changed).

Once this is done, original filenames lost, possible sequences of 2GB that go together for decoding (a bit like a multi file zip archive) lost.
I had only 4 that were forgotten in there before sending for import. So no big deal. But unexpected.

**Comment by graciousgrey** [comment]
> These files get imported, renamed (!?) in the Photoprism date format, but not indexed (ok) and get errors (ok) as files not recognized, no meaningful info. No yml created in sidecar (!? if file name changed).

That files are renamed during import is expected. If you want to keep your filenames you need to use indexing.

  https://docs.photoprism.app/user-guide/library/

If files cannot be indexed, no yml files are created. I fully agree that it would be helpful to display the original name in addition to the canonical name in the corresponding error messages. I have just created an issue for this:

- https://github.com/photoprism/photoprism/issues/3556

> These are old Magic Lantern raw video files. They basically contain a series of dng frames but probably encoded in a streaming (recording from sensor) format with headers if any that are unknown to Photoprism and its modules.

If you send us a sample file to samples@photoprism.app, we can look at whether it's possible to support this.

**Reply by artincube** [reply]
I know I need indexing to keep file names and organization. Basics. I have now about 50k items, imported for the undefined and copied manually and indexed for what was already organized.

The point is I expected Photoprism TO NOT import and change file names of files it doesn't understand. You see the idea? I am trying to find other sentence structure to get it through...
If no yml file is created for a media file because that media file cannot be recognized, then the original files should not be renamed or even moved from the import folder. Regardless of the possibility to index without importing.
I don't really care about original filename being displayed in error messages. I want error messages to mean that the file is not renamed neither moved from import folder and left there in its original folder structure, so I can have look at it in a quick and easy way.

Smallest sample file is 421MB. How do I send it?

**Reply by lastzero** [reply]
Files are only indexed once they are in the originals folder meaning many potential errors related to the file content (rather than just the checksum or the permissions) are unknown when they are copied or moved. Indexing them in the import folder would lead to non deterministic behavior and cause overhead as reindexing might be required. It would also interfere with how file stacks are recognized and indexed.

**Reply by artincube** [reply]
I understand. I was actually "afraid" of that. Well, now the only possibility would be to effectively import, but maybe sending back in case of errors that cannot be resolved, meaning files that can't get indexed at import get named back to their original names and sent back to the import folder. That would probably create other issues I suppose.

**Reply by lastzero** [reply]
We could do that, but it's the first time someone is requesting such a behavior and so I guess most users would rather want us to work on the features on our Roadmap. From what I know, an issue for improved logging has been created which shouldn't take too long to implement. With that, you could write a script to revert imports if the files have errors. Keep in mind that there are many complicated edge cases, like when you import a stack (e.g. Live Photo) and only one file out of many has problems. Would you then separate it? Or reject everything?

**Reply by artincube** [reply]
Good point. Also I understand that it is not common.
I usually import personal assets and index professional ones, because these are usually already organized, so this should not have happened. That is a very unusual situation that despite the confusion it created has little impact. Please focus on what has more.
Knowing that the logs are soon updated to include original filenames in case of errors/no yml created is something I can live with.
As you mentioned scripting is a possibility and I can do that.

**Reply by artincube** [reply]
Just for reference to keep about the files that created this:
Magic Lantern (RAW) Video format v2.0 (mlv_rec.mo)
https://www.magiclantern.fm/forum/index.php?topic=7122.0

The .raw was an earlier format, pre 2013.

**Reply by graciousgrey** [reply]
Thank you for the link! If you want to share a sample file with us you can send it to samples@photoprism.app
